Plugins/Memory/Collect-MemoryWinpmem.ps1

```diff
@@ -0,0 +1,194 @@
+            Continue
+        }
+
+        #Determine system architecture and select proper 7za.exe executable
+        try {
+         $Architecture = (Get-WmiObject -ComputerName $Computer -Class Win32_OperatingSystem -Property OSArchitecture -ErrorAction Stop).OSArchitecture
+        
+            if ($Architecture -eq "64-bit") {
+
+                $Installexe = $7za64
+
+            } elseif ($Architecture -eq "32-bit") {
+
+                $Installexe = $7za32
+
+            } else {
+            
+                Write-Error ("Unknown system architecture ({0}) detected for {1}. Data was not gathered.)" -f $Architecture, $Computer)
+                Continue
+            }
+
+        } catch {
+        
+         Write-Error ("Unable to determine system architecture for {0}. Data was not gathered." -f $Computer)
+            Continue
+        }
+
+        #Establish PS Session criteria
+
+        $Session = New-PSSession -ComputerName "$Computer"
+         
+        #Copy winpmem to remote host
+        $SmbPathWinpmem = ("\\{0}\c`$\ProgramData\{1}" -f $Computer, (Split-Path -Path $Winpmem -Leaf))
+        
+        try {
+        
+        Copy-Item -Path $Winpmem -Destination $SmbPathWinpmem -ErrorAction Stop
+        $RemoteFileWinpmem = Get-Item -Path $SmbPathWinpmem -ErrorAction Stop
+
+        # verify that the file copy succeeded to the remote host
+            if (-not $RemoteFileWinpmem) {
+            
+                Write-Error ("Winpmem not found on {0}. There may have been a problem during the copy process. Memory was not acquired." -f $Computer)
+                Continue
+
+                }
+
+            } catch {
+        
+        Write-Error ("An unexpected error occurred while copying winpmem to {0}. Memory not acquired." -f $Computer)
+        Continue
+
+        }
+
+        # Copy 7za.exe to remote system
+
+        $SmbPath7za = ("\\{0}\c`$\ProgramData\{1}" -f $Computer, (Split-Path -Path $Installexe -Leaf))
+        
+        try {
+        
+        Copy-Item -Path $Installexe -Destination $SmbPath7za -ErrorAction Stop
+        $RemoteFile7za = Get-Item -Path $SmbPath7za -ErrorAction Stop
+
+        # verify that the file copy succeeded to the remote host
+        if (-not $RemoteFile7za) {
+            
+            Write-Error ("7za.exe not found on {0}. There may have been a problem during the copy process. Memory cannot be compressed." -f $Computer)
+            Continue
+            }
+
+        } catch {
+        
+        Write-Error ("An unexpected error occurred while copying 7za.exe to {0}. Memory cannot be compressed." -f $Computer)
+        Continue
+
+        }
+
+        # Execute winpmem on remote machine to capture memory
+
+        $ScriptBlock_Mem = $ExecutionContext.InvokeCommand.NewScriptBlock(("& C:\ProgramData\{0} -o C:\ProgramData\{1}_memory.raw --volume_format raw -dd -t") -f ((Split-Path -Path $Winpmem -Leaf), $Computer))
```

Comment:
Execution errors out here for me with "PARSE ERROR Argument: -d (--debug), Argument already set!"

Reply:
Does your version work with `-dd` set? Or was this suppose to be `-d`?

Reply:
Apparently my version of winpmem doesn't take a mem capture if I remove the second d. It errors out complaining about `--volume-format`, the help page lists `--format` as a parameter

Seems more and more likely I have a different version of the tool. I got mine here: https://github.com/google/rekall/releases

Reply:
For winpmem -> https://github.com/Velocidex/c-aff4/releases/tag/3.2
Rekall doesnt have the most up to date winpmem... its being supported by michael coehen on the velocidex repo.

For 7zip -> https://www.7-zip.org/download.html (choose 7-Zip Extra: standalone console version, 7z DLL, Plugin for Far Manager)

You will need to rename the 7za.exe executables according to my script since they dont name themselves appropriately. one thing to consider would be to only use the 32 bit 7za.exe since it should work. I figured it would be best to use the x64 on x64 machines, but could be talked into making it simpler if that makes sense.

Overall, this part (downloading of stuff) would be put into the wiki to give a step by step walk through for dependencies.

Reply:
Looks good after having the right exes :)